[32mINFO    [0m integration.test_upgrade:test_upgrade.py:158 starting continuous writes to the database
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:162 checking whether writes are increasing
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:165 Get leader unit
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:171 Run pre-upgrade-check action
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:179 Inject dependency fault
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:184 Refresh the charm
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:187 Get first upgrading unit
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:191 Wait for upgrade to fail on first upgrading unit
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:198 Ensure continuous_writes while in failure state on remaining units
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:201 Re-run pre-upgrade-check action
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:205 Re-refresh the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer-k8s/0 [idle] waiting: other units upgrading first...
  pgbouncer-k8s/1 [idle] waiting: other units upgrading first...
  pgbouncer-k8s/2 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer-k8s/0 [idle] waiting: other units upgrading first...
  pgbouncer-k8s/1 [idle] active: 
  pgbouncer-k8s/2 [idle] active: 
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:212 Resume upgrade
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:216 Wait for application to recover
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer-k8s/0 [idle] waiting: other units upgrading first...
  pgbouncer-k8s/1 [executing] maintenance: 
  pgbouncer-k8s/2 [idle] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer-k8s/0 [executing] maintenance: 
  pgbouncer-k8s/1 [idle] active: 
  pgbouncer-k8s/2 [idle] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer-k8s/0 [idle] active: 
  pgbouncer-k8s/1 [idle] active: 
  pgbouncer-k8s/2 [idle] active: 
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:220 Ensure continuous_writes after rollback procedure
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:225 Checking whether no writes were lost
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-1 "HTTP/1.1 200 OK"
[32mINFO    [0m httpx:_client.py:1025 HTTP Request: GET https://10.1.0.21:16443/api/v1/namespaces/testing/pods/postgresql-k8s-2 "HTTP/1.1 200 OK"
[32mINFO    [0m pytest_operator.plugin:plugin.py:936 Model status:

Model    Controller          Cloud/Region        Version  SLA          Timestamp
testing  concierge-microk8s  microk8s/localhost  3.6.3    unsupported  01:58:28Z

App                  Version  Status  Scale  Charm                Channel      Rev  Address         Exposed  Message
pgbouncer-k8s        1.21.0   active      3  pgbouncer-k8s                       2  10.152.183.25   no       
postgresql-k8s       14.15    active      3  postgresql-k8s       14/edge      509  10.152.183.87   no       
postgresql-test-app           active      1  postgresql-test-app  latest/edge  365  10.152.183.229  no       received database credentials of the first database

Unit                    Workload  Agent  Address       Ports  Message
pgbouncer-k8s/0         active    idle   10.1.254.153         
pgbouncer-k8s/1         active    idle   10.1.254.152         
pgbouncer-k8s/2*        active    idle   10.1.254.151         
postgresql-k8s/0*       active    idle   10.1.254.144         Primary
postgresql-k8s/1        active    idle   10.1.254.145         
postgresql-k8s/2        active    idle   10.1.254.146         
postgresql-test-app/0*  active    idle   10.1.254.137         received database credentials of the first database

[32mINFO    [0m pytest_operator.plugin:plugin.py:942 Juju error logs:

unit-postgresql-k8s-2: 01:49:15 ERROR unit.postgresql-k8s/2.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-1: 01:49:15 ERROR unit.postgresql-k8s/1.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-2: 01:49:45 ERROR unit.postgresql-k8s/2.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-pgbouncer-k8s-2: 01:52:24 ERROR unit.pgbouncer-k8s/2.juju-log pgbouncer not running
unit-pgbouncer-k8s-2: 01:52:27 ERROR unit.pgbouncer-k8s/2.juju-log pgbouncer not running
unit-pgbouncer-k8s-1: 01:53:24 ERROR unit.pgbouncer-k8s/1.juju-log pgbouncer not running
unit-pgbouncer-k8s-1: 01:53:25 ERROR unit.pgbouncer-k8s/1.juju-log pgbouncer not running
unit-pgbouncer-k8s-1: 01:53:27 ERROR unit.pgbouncer-k8s/1.juju-log pgbouncer not running
unit-pgbouncer-k8s-1: 01:53:28 ERROR unit.pgbouncer-k8s/1.juju-log pgbouncer not running
unit-pgbouncer-k8s-0: 01:53:47 ERROR unit.pgbouncer-k8s/0.juju-log pgbouncer not running
unit-pgbouncer-k8s-0: 01:53:49 ERROR unit.pgbouncer-k8s/0.juju-log pgbouncer not running
unit-pgbouncer-k8s-0: 01:53:51 ERROR unit.pgbouncer-k8s/0.juju-log pgbouncer not running
unit-pgbouncer-k8s-0: 01:53:52 ERROR unit.pgbouncer-k8s/0.juju-log pgbouncer not running
unit-pgbouncer-k8s-2: 01:55:33 ERROR unit.pgbouncer-k8s/2.juju-log charms.data_platform_libs.v0.upgrade.VersionError - {'message': 'Versions incompatible, charm 1 can not be upgraded to 15.0', 'cause': 'Upgrades only supported for specific versions, charm versions satisfying requirement ^15', 'resolution': ''}
unit-pgbouncer-k8s-2: 01:55:35 ERROR unit.pgbouncer-k8s/2.juju-log pgbouncer not running
unit-pgbouncer-k8s-2: 01:55:39 ERROR unit.pgbouncer-k8s/2.juju-log pgbouncer not running
unit-pgbouncer-k8s-1: 01:57:14 ERROR unit.pgbouncer-k8s/1.juju-log pgbouncer not running
unit-pgbouncer-k8s-1: 01:57:16 ERROR unit.pgbouncer-k8s/1.juju-log pgbouncer not running
unit-pgbouncer-k8s-1: 01:57:17 ERROR unit.pgbouncer-k8s/1.juju-log pgbouncer not running
unit-pgbouncer-k8s-1: 01:57:19 ERROR unit.pgbouncer-k8s/1.juju-log pgbouncer not running
unit-pgbouncer-k8s-0: 01:57:38 ERROR unit.pgbouncer-k8s/0.juju-log pgbouncer not running
unit-pgbouncer-k8s-0: 01:57:40 ERROR unit.pgbouncer-k8s/0.juju-log pgbouncer not running
unit-pgbouncer-k8s-0: 01:57:41 ERROR unit.pgbouncer-k8s/0.juju-log pgbouncer not running
unit-pgbouncer-k8s-0: 01:57:42 ERROR unit.pgbouncer-k8s/0.juju-log pgbouncer not running

[32mINFO    [0m pytest_operator.plugin:plugin.py:1024 Forgetting model main...